docs/src/Guides/20 Extensions.md

```diff
@@ -1,248 +1,581 @@
-To do this, simply run `bot.reload_extension("Filename_here")` and your new code will be used. Bare in mind any tasks your extension
-is doing will be abruptly stopped.
+To load the extension, you just need to use `bot.load_extension("filename.in_import_style")` before `bot.start`. So, in this case, it would look like this:
+
+```python
+bot = Client(...)
+bot.load_extension("extension")
+bot.start("token")
+```
+
+And that's it! Your extension is now loaded and ready to go.
+
+#### "Import Style"
+
+In the example above, the filename is passed to `load_extension` without the `.py` extension. This is because interactions.py actually does an
+*import* when loading the extension, so whatever string you give it needs to be a valid Python import path. This means that if you have a file structure like this:
+
+```
+main.py
+exts/
+    extension.py
+```
+
+You would need to pass `exts.extension` to `load_extension`, as that's the import path to the extension file.
 
+#### Reloading and Unloading Extensions
 
-You can pass keyword-arguments to the `load_extension`, `unload_extension` and `reload_extension` extension management methods.
-Any arguments you pass to the `setup` or `teardown` methods, will also be passed to the `Extension.drop` method.
+You can also reload and unload extensions. To do this, you use `bot.reload_extension` and `bot.unload_extension` respectively.
+
+```python
+bot.reload_extension("extension")
+bot.unload_extension("extension")
+```
 
-Here is a basic "Extension switching" example:
+Reloading and unloading extensions allows you to edit your code without restarting the bot, and to remove extensions you no longer need.
+For example, if you organize your extensions so that moderation commands are in one extension, you can reload that extension (and so only moderation-related commands)
+as you edit them.
+
+### Initialization
+
+You may want to do some logic to do when loading a specific extension. For that, you can add the `__init__` method, which takes a `Client` instance, in your extension:
 
 ```python
-from interactions import Extension
+class MyExtension(Extension):
+    def __init__(self, bot):
+        # do some initialization here
+        pass
+```
+
+#### Asynchronous Initialization
+
+As usual, `__init__` is synchronous. This may pose problems if you're trying to do something asynchronous in it, so there are various ways of solving it.
+
+If you're okay with only doing the asynchronous logic as the bot is starting up (and never again), there are two methods:
+
+=== "`async_start`"
+    ```python
+    class MyExtension(Extension):
+        async def async_start(self):
+            # do some initialization here
+            pass
+    ```
+
+=== "`Startup` Event"
+    ```python
+    from interactions.api.events import Startup
+
+    class MyExtension(Extension):
+        @event(Startup)
+        async def startup(self):
```

Comment:
If the startup declaration is optional for non-async, should we clarify how it's used here?

Reply:
I mean, it's noted as being one of many options you can use in the first place, I think that's good enough.